openedx_certificates/renderers/elements.py

```diff
@@ -0,0 +1,153 @@
+        attributes.get('x', 0),
+        attributes.get('y', 0),
+        attributes.get('width', 0),
+        attributes.get('height', 0),
+        mask='auto',
+    )
+
+
+def draw_text(certificateGen, attributes, canvas, context):
+    string = unicode(attributes.get("string", ""))
+
+    if context:
+        string = string.format(**context)
+
+    fontSize = attributes.get("font_size", 12)
+    leading = attributes.get("leading", 12)
+    textColor = attributes.get("text_color", [0, 0, 0])
+    height = attributes.get("height", 0)
+    width = attributes.get("width", 0)
+    x_position = attributes.get("x", 0)
+    y_position = attributes.get("y", 0)
+    alignment = attributes.get("alignment", "left")
+    alignment = _translate_reportlab_alignment(alignment)
+    auto_scale = attributes.get("auto_scale")
+
+    style_for_text = ParagraphStyle(
+        name="text",
+        fontSize=fontSize,
+        leading=leading,
+        textColor=textColor,
+        alignment=getattr(reportlab_enums, alignment),
+    )
+
+    (fonttag, fontfile, text_style) = font_for_string(
+        apply_style_to_font_list(certificateGen.fontlist, style_for_text),
+        string
+    )
+
+    max_height = height
+    if auto_scale:
+        paragraph = autoscale_text(canvas, string, fontSize, leading, height, width, text_style)
+        y_position = y_position + ((height - paragraph.height) / 2) + (text_style.fontSize / 5)
+    else:
+        paragraph = Paragraph(string, text_style)
+        width, height = paragraph.wrapOn(canvas, width, height)
+        y_position = y_position + (max_height - paragraph.height)
+
+    paragraph.drawOn(canvas, x_position, y_position)
+
+
+element_options = {
+    "rectangle": draw_rect,
+    "line": draw_line,
+    "image": draw_image,
+    "text": draw_text,
+}
+
+
+def draw_template_element(certificateGen, element, attributes, canvas, context=None):
+    element_options[element](certificateGen, attributes, canvas, context)
+
+
+def draw_instructor_element(certificateGen, instructor_attributes, instructor_template, canvas):
+    x_position = instructor_attributes.get("x_position", 0)
+    y_position = instructor_attributes.get("y_position", 0)
+    for ordered_element in instructor_template:
+        ordered_element_copy = copy.deepcopy(ordered_element)
+        for element, attributes in ordered_element_copy.iteritems():
+            x_offset = attributes.get('x_offset', 0)
+            if element == "text":
+                attributes['string'] = instructor_attributes.get(attributes.get("string", ""), "")
+                attributes['x'] = x_position + x_offset
+                attributes['y'] = y_position
+                if attributes['string']:
+                    y_position += attributes.get('height', 0)
+                else:
+                    continue
+            elif element == "image":
+                attributes['x'] = x_position + x_offset
+                attributes['y'] = y_position - 10
```

Comment:
move as first line in `if attributes['file']:` and calculate image height using: http://stackoverflow.com/questions/5327670/image-aspect-ratio-using-reportlab-in-python